python/src/aiconfig/editor/client/src/components/prompt/PromptMenuButton.tsx

```diff
@@ -10,13 +11,27 @@ const useStyles = createStyles(() => ({
 
 export default memo(function PromptMenuButton({
   promptId,
+  prompt,
+  onDeleteOutput,
   onDeletePrompt,
 }: {
   promptId: string;
+  prompt: ClientPrompt;
```

Comment:
Passing `prompt` through just to access the output length will mean this component is re-rendered any time any part of the prompt changes (e.g. if typing in the input). That will result in the React memo constantly re-calculating the diff to check if it should re-render, and also re-rendering unnecessarily. Instead, it would be better to have a prop `showDeleteOutput` which is set to true if the prompt.outputs.length > 0 in the parent component